lightning/src/routing/network_graph.rs

```diff
@@ -237,6 +282,349 @@ impl<C: Deref + Sync + Send, L: Deref + Sync + Send> RoutingMessageHandler for N
 			false
 		}
 	}
+
+	fn query_channel_range(&self, their_node_id: &PublicKey, chain_hash: BlockHash, first_blocknum: u32, number_of_blocks: u32) -> Result<(), LightningError> {
+		// We must ensure that we only have a single in-flight query
+		// to the remote peer. If we already have a query, then we fail
+		let mut query_range_tasks_lock = self.query_channel_range_tasks.lock().unwrap();
+		let query_range_tasks = &mut *query_range_tasks_lock;
+		if query_range_tasks.contains_key(their_node_id) {
+			return Err(LightningError {
```

Comment:
If it's a caller requirement to track channel_range query I think this might be a hint of a higher logic bug. Though I don't have a clear model of this method usage.

Reply:
My thinking is that this will be called from peer_handler after init negotiation and `should_request_full_sync` returns true. This guard prevents multiple in-flight `query_channel_range` messages which is spec non-compliant. This could be changed to queue channel range queries but I thought outright rejection was a simpler for now.  

Relatedly, my plan is to make `full_syncs_requested` decrement on any failure in `handle_reply_channel_range` so that a full sync can be requested with the next peer.

Reply:
Right, we can switch it later to an assert when higher sync logic is implemented.